website/static/js/dashboard.js

```diff
@@ -0,0 +1,1187 @@
+            promise.then(function(result){
+                for ( var i = 0; i < self.collections().length; i++) {
+                    var item = self.collections()[i];
+                    if (item.data.node && item.data.node.id === self.collectionMenuObject().item.data.node.id){
+                        self.collections().splice(i, 1);
+                        break;
+                    }
+                }
+            });
+            self.dismissModal();
+            self.calculateTotalPages();
+            return promise;
+        };
+        self.renameCollection = function _renameCollection() {
+            var url = self.collectionMenuObject().item.data.node.links.self;
+            var nodeId = self.collectionMenuObject().item.data.node.id;
+            var title = self.collectionMenuObject().item.renamedLabel;
+            var data = {
+                'data': {
+                    'type': 'collections',
+                    'id':  nodeId,
+                    'attributes': {
+                        'title': title
+                    }
+                }
+            };
+            var promise = m.request({method : 'PATCH', url : url, config : xhrconfig, data : data});
+            promise.then(function(result){
+                var updatedCollMenuObj = $.extend(true, {}, self.collectionMenuObject(), {item: {label: title}});
+                self.collectionMenuObject(updatedCollMenuObj);
+            });
+            self.dismissModal();
+            self.validation(false);
+            return promise;
+        };
+        self.applyDroppable = function _applyDroppable ( ){
+            $('.db-collections ul>li').droppable({
+                hoverClass: 'bg-color-hover',
+                drop: function( event, ui ) {
+                    var collection = self.collections()[$(this).attr('data-index')];
+                    var dataArray = [];
+                    // If multiple items are dragged they have to be selected to make it work
+                    if (args.selected().length > 1) {
+                        args.selected().map(function(item){
+                            dataArray.push(buildCollectionNodeData(item.data.id));
+                        });
+                    } else {
+                        // if single items are passed use the event information
+                        dataArray.push(buildCollectionNodeData(ui.draggable.find('.title-text>a').attr('data-nodeID'))); // data-nodeID attribute needs to be set in project organizer building title column
+                    }
+                    function saveNodetoCollection (index) {
+                        function doNext (){
+                            if(dataArray[index+1]){
+                                saveNodetoCollection(index+1);
+                            }
+                            collection.data.count(collection.data.count()+1);
+                        }
+                        m.request({
```

Comment:
@pattisdr Is there a way we can use the Bulk API to add multiple nodes to a collection in a single request, rather than sending multiple requests as we're doing here?